events/listener.go

```diff
@@ -35,3 +49,18 @@ func Listen(eventURL, accessKey, secretKey string, workerCount int) error {
 	err = router.StartWithoutCreate(nil)
 	return err
 }
+
+func checkTS(timestamps *time.Time) bool {
+	stampFile := config.Stamp()
+	if !utils.IsPathExist(stampFile) {
+		return true
+	}
+	// ignore the err cause the file exists
+	stats, _ := os.Stat(stampFile)
+	ts := stats.ModTime()
+	// check whether timestamps has been initialized
+	if timestamps.IsZero() {
+		timestamps = &ts
```

Comment:
I think this is doing the opposite of what you want.  You are changing the local pointer, not setting the point to the new value.  I think you want

```go
*timestamp = ts
```